### PR TITLE
Credibility check estimation

### PR DESCRIPTION
**Pull request by AlexeyPichugin**
Added credibility check estimation

### CODE REVIEW

File: lunavl/sdk/estimators/face_estimators/credibility_check.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    def credibilityCheck(self) -> float:
+        """
+        The credibility check
+
+        Returns:
+            float in range(0, 1)
+        """
+        return self._coreEstimation.value
+
+    def asDict(self) -> Union[Dict[Any, Any]]:
```
Comment (DeusAnimo) [inline lunavl/sdk/estimators/face_estimators/credibility_check.py:45]: wrong typing

File: lunavl/sdk/estimators/face_estimators/credibility_check.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+from FaceEngine import ICredibilityCheckEstimatorPtr
+from FaceEngine import CredibilityCheckEstimation
+
+from lunavl.sdk.errors.errors import LunaVLError
+from lunavl.sdk.errors.exceptions import CoreExceptionWrap, LunaSDKException
+from ..base import BaseEstimator
+from ..face_estimators.facewarper import FaceWarp, FaceWarpedImage
+from ...base import BaseEstimation
+
+from typing import Union, Dict, Any
```
Comment (DeusAnimo) [inline lunavl/sdk/estimators/face_estimators/credibility_check.py]: move up typing

File: lunavl/sdk/estimators/face_estimators/credibility_check.py (hunk 1 above)
Comment (DeusAnimo) [inline lunavl/sdk/estimators/face_estimators/credibility_check.py]: wrong typing

File: lunavl/sdk/estimators/face_estimators/credibility_check.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    #  pylint: disable=W0235
+    def __init__(self, credibilityCheck: CredibilityCheckEstimation):
+        """
+        Init.
+
+        Args:
+            credibilityCheck: estimated credibility check
+        """
+        super().__init__(credibilityCheck)
```
Comment (DeusAnimo) [inline lunavl/sdk/estimators/face_estimators/credibility_check.py]: why `super`? remove this init

File: lunavl/sdk/estimators/face_estimators/credibility_check.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        """
+        return {"credibility_check": self.credibilityCheck}
+
+
+class CredibilityCheckEstimator(BaseEstimator):
+    """
+    Warp credibility check estimator.
+    """
+
+    def __init__(sekf, credibilityCheckEstimator: ICredibilityCheckEstimatorPtr):
```
Comment (DeusAnimo) [inline lunavl/sdk/estimators/face_estimators/credibility_check.py]: sekf ? remove excess initializetion

File: tests/test_estimate_descriptor.py [hunk 5]
```diff
@@ -439,11 +439,9 @@ def test_descriptor_batch_bad_threshold_aggregation(self):
...
-                self.assertLunaVlError(
-                    exceptionInfo,
-                    LunaVLError.BatchedInternalError.format(
-                        "Cant aggregate descriptors - all images'a GSs are less the threashold"
-                    ),
+                descriptorsRaw, descriptorAggregated = extractor.estimateDescriptorsBatch(
+                    [faceWarp] * 2, aggregate=1, descriptorBatch=descriptorBatch
                 )
+                for descriptorRaw in descriptorsRaw:
+                    self.assertDescriptor(descriptorVersion, descriptorRaw, DescriptorType.face)
```
Comment (DeusAnimo) [inline tests/test_estimate_descriptor.py:446]: why is only one version of the descriptor used?

File: tests/test_credibility_check.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+    @classmethod
+    def setup_class(cls):
+        super().setup_class()
+        cls.credibilityCheckEstimator = cls.faceEngine.createCredibilityCheckEstimator()
+
+        cls.warp = FaceWarpedImage(VLImage.load(filename=WARP_CLEAN_FACE))
+
+    def assertrCedibilityCheckEstimation(
+        self, credibilityCheck: CredibilityCheck, expectedEstimationResults: Dict[str, str]
```
Comment (DeusAnimo) [inline tests/test_credibility_check.py]: expectedEstimationResults - wrong typing

File: tests/test_credibility_check.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        """
+        Function checks if the instance belongs to the credibility check class 
+        and compares the result with what is expected.
+
+        Args:
+            credibilityCheck: credibility check estimation object
+            expectedEstimationResults: dictionary with result
+        """
+        assert isinstance(credibilityCheck, CredibilityCheck), f"{credibilityCheck.__class__} is not {CredibilityCheck}"
+        credibilityCheckScore = credibilityCheck.asDict()["credibility_check"]
```
Comment (DeusAnimo) [inline tests/test_credibility_check.py]: you have a class method

File: tests/test_credibility_check.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    ):
+        """
+        Function checks if the instance belongs to the credibility check class 
+        and compares the result with what is expected.
+
+        Args:
+            credibilityCheck: credibility check estimation object
+            expectedEstimationResults: dictionary with result
+        """
+        assert isinstance(credibilityCheck, CredibilityCheck), f"{credibilityCheck.__class__} is not {CredibilityCheck}"
```
Comment (DeusAnimo) [inline tests/test_credibility_check.py:35]: add check for result type (`isinstance() -> float`) and value in range [0,1]

File: tests/test_credibility_check.py [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+from lunavl.sdk.estimators.face_estimators.facewarper import FaceWarpedImage
+from lunavl.sdk.estimators.face_estimators.credibility_check import CredibilityCheck, CredibilityCheckEstimator
+from lunavl.sdk.image_utils.image import VLImage
+
+from tests.base import BaseTestClass
+from tests.resources import WARP_CLEAN_FACE
+from tests.schemas import jsonValidator, CREDIBILITY_CHECK_SCHEMA
+
+
+class TestCredibilityCheck(BaseTestClass):
```
Comment (DeusAnimo) [inline tests/test_credibility_check.py:12]: I don't see tests with bad cases

File: tests/test_credibility_check.py (hunk 8 above)
Comment (AlexeyPichugin) [reply tests/test_credibility_check.py:35]: Doesn't this check 'CREDIBILITY_CHECK_SCHEMA' schema? There, in the properties is used TYPE_SCORE, which checks the type and limit of the value

File: lunavl/sdk/estimators/face_estimators/credibility_check.py (hunk 3 above)
Comment (AlexeyPichugin) [reply lunavl/sdk/estimators/face_estimators/credibility_check.py]: Why should I remove the init statement? It sets the `_coreEstimation`, which is used to get the estimated value.

File: tests/test_credibility_check.py (hunk 7 above)
Comment (AlexeyPichugin) [reply tests/test_credibility_check.py]: Done

File: tests/test_credibility_check.py (hunk 6 above)
Comment (AlexeyPichugin) [reply tests/test_credibility_check.py]: Fixed

File: lunavl/sdk/estimators/face_estimators/credibility_check.py (hunk 4 above)
Comment (AlexeyPichugin) [reply lunavl/sdk/estimators/face_estimators/credibility_check.py]: Fixed

File: lunavl/sdk/estimators/face_estimators/credibility_check.py (hunk 1 above)
Comment (AlexeyPichugin) [reply lunavl/sdk/estimators/face_estimators/credibility_check.py]: Fixed

File: lunavl/sdk/estimators/face_estimators/credibility_check.py (hunk 2 above)
Comment (AlexeyPichugin) [reply lunavl/sdk/estimators/face_estimators/credibility_check.py]: Fixed

File: tests/test_estimate_descriptor.py (hunk 5 above)
Comment (AlexeyPichugin) [reply tests/test_estimate_descriptor.py:446]: added all descriptor versions

File: tests/test_credibility_check.py (hunk 8 above)
Comment (DeusAnimo) [reply tests/test_credibility_check.py:35]: the numeric type for json schema is not equal a python type object

File: lunavl/sdk/estimators/face_estimators/credibility_check.py (hunk 3 above)
Comment (AlexeyPichugin) [reply lunavl/sdk/estimators/face_estimators/credibility_check.py]: fixed

File: tests/test_credibility_check.py (hunk 8 above)
Comment (AlexeyPichugin) [reply tests/test_credibility_check.py:35]: fixed

**Pull request merged**